OpenRA.Mods.Common/Widgets/Logic/Ingame/ProductionTooltipLogic.cs

```diff
@@ -70,8 +70,12 @@ public ProductionTooltipLogic(Widget widget, TooltipContainerWidget tooltipConta
 				var tooltip = actor.TraitInfos<TooltipInfo>().FirstOrDefault(info => info.EnabledByDefault);
 				var name = tooltip != null ? tooltip.Name : actor.Name;
 				var buildable = actor.TraitInfo<BuildableInfo>();
-				var valued = actor.TraitInfoOrDefault<ValuedInfo>();
-				var cost = valued != null ? valued.Cost : 0;
+
+				var cost = 0;
+				if (tooltipIcon.ProductionQueue != null)
+					cost = tooltipIcon.ProductionQueue.GetProductionCost(actor);
+				else if (actor.TraitInfoOrDefault<ValuedInfo>() != null)
```

Comment:
How can we have a tooltip if we don't have a production queue?

If we do need this fallback, please reduce it to a single query using
```csharp
else
{
	var valued = actor.TraitInfoOrDefault<ValuedInfo>();
	if (valued != null)
		cost = valued.Cost;
}